pkg/standalone-utils/contracts/TokenPairRegistry.sol

```diff
@@ -178,22 +175,20 @@ contract TokenPairRegistry is ITokenPairRegistry, OwnableAuthentication {
 
         IBatchRouter.SwapPathStep[][] storage paths = _pairsToPaths[tokenId];
 
-        // We first look for a path of length 1 which is the one for a simple pair step.
+        // We first look for a path of length 1, representing a simple pair.
         for (uint256 i = 0; i < paths.length; ++i) {
             IBatchRouter.SwapPathStep[] storage steps = paths[i];
-            if (steps.length > 1) {
-                continue;
-            }
 
-            if (steps[0].pool == poolOrBuffer && address(steps[0].tokenOut) == tokenOut) {
-                // Element found. Re-arrange paths if needed, swapping last element with the current one.
+            if (steps.length == 1 && steps[0].pool == poolOrBuffer && address(steps[0].tokenOut) == tokenOut) {
```

Comment:
That's fair, although steps.length shouldn't be 0 at this point

Reply:
Mainly just reducing one level of nesting; also continues take a surprising amount of bytecode. Doubles as a check to ensure it's not zero (which as you point out, should be impossible anyway); otherwise it would revert with an array index out of range